src/main/java/org/folio/qm/converter/MarcQmConverter.java

```diff
@@ -61,12 +66,36 @@ private JsonNode convertToParsedContent(T source, MarcFormat format) {
     try (ByteArrayOutputStream os = new ByteArrayOutputStream();
          QmMarcJsonWriter writer = new QmMarcJsonWriter(os)) {
       writer.write(marcRecord);
-      return objectMapper.readTree(os.toByteArray());
+
+      var convertedContent = objectMapper.readTree(os.toByteArray());
+      reorderContentTagsBasedOnSource(convertedContent, source.getFields());
+      return convertedContent;
     } catch (IOException e) {
       throw new ConverterException(e);
     }
   }
 
+  private void reorderContentTagsBasedOnSource(JsonNode jsonNode, List<FieldItem> sourceFields) {
+    var fieldsArrayNode = (ArrayNode) jsonNode.path("fields");
+
+    Map<String, Queue<JsonNode>> jsonNodesByTag = new HashMap<>();
+    fieldsArrayNode.forEach(node -> {
+      String tag = node.fieldNames().next();
+      jsonNodesByTag.computeIfAbsent(tag, k -> new LinkedList<>()).add(node);
+    });
+
+    var rearrangedArray = objectMapper.createArrayNode();
+    for (FieldItem fieldItem : sourceFields) {
+      Queue<JsonNode> nodes = jsonNodesByTag.get(fieldItem.getTag());
+      if (nodes != null && !nodes.isEmpty()) {
+        rearrangedArray.add(nodes.poll());
+      }
+    }
+
+    fieldsArrayNode.removeAll();
+    fieldsArrayNode.addAll(rearrangedArray);
+  }
+
```

Comment:
I don't think it will work fine due to tags could be repeatable and tags with same number could be placed in different place of an array by user, With this approach all these repeatable tags will be placed in one place near each other.

Reply:
functionality checked and verified, works as expected.